ride-share.rb

```diff
@@ -0,0 +1,249 @@
+# Which layers are nested in each other?
+    # drivers > rides > ride_data  (date, cost, rider_id, rating)
+
+# Which layers of data "have" within it a different layer?
+    # drivers, and rides, and ride_data all have an additional data structure within them
+      # drivers has rides, which is an array
+      # ride has ride data, which is a hash
+      # rides has data with the structure of key => value
+
+# Which layers are "next" to each other?
+    # The data within ride_data has data next to one another -  date, cost, rider_id, rating are all next to one another in the same hash
+    # The drivers are next to one another within a hash
+    # The rides are next to one another within an array
+
+########################################################
+# Step 2: Assign a data structure to each layer
+
+# Copy your list from above, and in this section
+# determine what data structure each layer should have
+    # drivers [hash] > rides [array] > ride_data [hash]
+
+########################################################
+# Step 3: Make the data structure!
+
+# Setup the entire data structure:
+# based off of the notes you have above, create the
+# and manually write in data presented in rides.csv
+# You should be copying and pasting the literal data
+# into this data structure, such as "DR0004"
+# and "3rd Feb 2016" and "RD0022"
+
+drivers =
+{
+  DR0001:
+  [
+      {
+        date: "3rd Feb 2016",
+        cost: 30,
+        rider_id: "RD0015",
+        rating: 4
+      },
+
+      {
+        date: "3rd Feb 2016",
+        cost: 10,
+        rider_id: "RD0003",
+        rating: 3
+      },
+
+      {
+        date: "5th Feb 2016",
+        cost: 45,
+        rider_id: "RD0003",
+        rating: 2
+      },
+  ],
+
+  DR0002:
+  [
+    {
+      date: "3rd Feb 2016",
+      cost: 25,
+      rider_id: "RD0073",
+      rating: 5
+    },
+
+    {
+      date: "4th Feb 2016",
+      cost: 15,
+      rider_id: "RD0013",
+      rating: 1
+    },
+
+    {
+      date: "5th Feb 2016",
+      cost: 35,
+      rider_id: "RD0066",
+      rating: 3
+    },
+
+  ],
+
+  DR0003:
+  [
+    {
+      date: "4th Feb 2016",
+      cost: 5,
+      rider_id: "RD0066",
+      rating: 5
+    },
+
+    {
+      date: "5th Feb 2016",
+      cost: 50,
+      rider_id: "RD0003",
+      rating: 2
+    },
+
+  ],
+
+  DR0004:
+  [
+    {
+      date: "3rd Feb 2016",
+      cost: 5,
+      rider_id: "RD0022",
+      rating: 5
+    },
+
+    {
+      date: "4th Feb 2016",
+      cost: 10,
+      rider_id: "RD0022",
+      rating: 4
+    },
+
+    {
+      date: "5th Feb 2016",
+      cost: 20,
+      rider_id: "RD0073",
+      rating: 5
+    },
+
+  ],
+
+}
+
+########################################################
+# Step 4: Total Driver's Earnings and Number of Rides
+
+# Use an iteration blocks to print the following answers:
+
+# 1
+puts "Number of rides each driver has given: "
+drivers.each do |key, value|
+  puts "   #{key} has given #{value.length} rides"
+end
```

Comment:
Are there more descriptive names you could use for these block parameters than `key` and `value`? Something like `driver_id` and `trips` might be a good choice.